### DISCUSSION TITLE
[react-markdown] Use react markdown in newly generated content

### DISCUSSION THREAD

**Discussion by hilmanski** [remarkjs/remark#800] [Q&A] [answered]
Hi 

I'd like to ask, is there any way to use react-markdown in new dynamic generated content.

The case is I want to edit a post and return the result from DB.
Result is a text with markdown that I want to display directly, with react-markdown.
Not sure how to do it  inside a function. (Not on inital render)

something like this
document.querySelector(`#myDiv`).innerText = <ReactMarkdown>(data.reply.body)</ReactMarkdown>
//definitely not working, since it's not a render function

**Comment by ChristianMurphy** [comment, marked as answer]
Make the markdown text/content a piece of state.
Give it an empty value or a loader by default.
Have the database call update the state when the load is complete.
Then the component can be plain JSX, no manual/unmanaged DOM manipulation needed.

https://reactjs.org/docs/hooks-state.html

**Reply by hilmanski** [reply]
thanks for replying @ChristianMurphy 
it works!
